src/subtitles/subtitles.js

```diff
@@ -8,14 +8,20 @@ function Subtitles(mediaPlayer, autoStart, playbackElement, defaultStyleOpts, me
   let subtitlesContainer
 
   if (useLegacySubs) {
-    import("./legacysubtitles.js")
-      .then(({ default: LegacySubtitles }) => {
-        subtitlesContainer = LegacySubtitles(mediaPlayer, autoStart, playbackElement, mediaSources, defaultStyleOpts)
+    if (available()) {
+      import("./legacysubtitles.js")
+        .then(({ default: LegacySubtitles }) => {
+          subtitlesContainer = LegacySubtitles(mediaPlayer, autoStart, playbackElement, mediaSources, defaultStyleOpts)
+          callback(subtitlesEnabled)
+        })
+        .catch(() => {
+          Plugins.interface.onSubtitlesDynamicLoadError()
+        })
+    } else {
+      new Promise((resolve) => resolve()).then(() => {
```

Comment:
Could you add a comment to explain why this is necessary? Maybe link https://developer.mozilla.org/en-US/docs/Web/API/HTML_DOM_API/Microtask_guide#using_microtasks

Could you simplify with `Promise.resolve(() => callback())` or `Promise.resolve().then(callback)`?

Reply:
eb003ed